go-controller/pkg/ovn/gateway_init.go

```diff
@@ -391,15 +412,45 @@ func (oc *Controller) gatewayInit(nodeName string, clusterIPSubnet []*net.IPNet,
 	// if config.Gateway.DisabledSNATMultipleGWs is not set (by default it is not),
 	// the NAT rules for pods not having annotations to route through either external
 	// gws or pod CNFs will be added within pods.go addLogicalPort
+	externalIPs := make([]net.IP, len(l3GatewayConfig.IPAddresses))
+	for i, ip := range l3GatewayConfig.IPAddresses {
+		externalIPs[i] = ip.IP
+	}
+	var natsToUpdate []*nbdb.NAT
+	// If l3gatewayAnnotation.IPAddresses changed, we need to update the SNATs on the GR
+	if len(oldExtIPs) > 0 {
+		for _, logicalSubnet := range clusterIPSubnet {
```

Comment:
this is odd that you are looping over clusterIPSubnets for setting the external IP on a NAT. I think you can just loop over the external ips, then look for same family in the oldExternalIPs

Reply:
right but we are trying to get the correct family from the l3 annotation for the pods right? So depending on the family of clusterSubnet I need to pull the ipv4 or ipv6 externalIP from the annotation list.

Reply:
oh wait nvm I see what you mean :) . yeah what you are saying is way better